File: codeforces/gym/102951/A.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

const int INF = 100000000;

vector<int> get_list_from_line() {
    string line;
    getline(cin, line);
    vector<int> result;
    int num = 0;
    for (char c : line) {
        if (isdigit(c)) {
            num = num * 10 + (c - '0');
        } else {
            result.push_back(num);
            num = 0;
        }
    }
    if (num != 0) {
        result.push_back(num);
    }
    return result;
}

int main() {
    int N;
    cin >> N;
    //    cin.ignore(); // Ignore the newline character after reading N
    
    //    vector<int> x_list = get_list_from_line();
    //    vector<int> y_list = get_list_from_line();
    vector<int> x_list;
    vector<int> y_list;
    int num;
    for (int i = 0; i < N; ++i) {
      cin >> num;
      x_list.push_back(num);
    }
    for (int i = 0; i < N; ++i) {
      cin >> num;
      y_list.push_back(num);
    }

    int best = 0;
    for (int i = 0; i < N; ++i) {
        for (int j = i + 1; j < N; ++j) {
            int w = x_list[i] - x_list[j];
            int h = y_list[i] - y_list[j];
            best = max(best, h * h + w * w);
        }
    }

    cout << best << endl;

    return 0;
}

File: codeforces/gym/102951/A.py
import sys
INF = 100000000


def get_list_from_line(dtype=int):
    return [dtype(x) for x in fin.readline().strip().split()]


def get_strs_from_line():
    return fin.readline().strip().split()



fin = sys.stdin
fout = sys.stdout

N = get_list_from_line()[0]
x_list = get_list_from_line()
y_list = get_list_from_line()

best = 0
for i in range(N):
    for j in range(i + 1, N):
        w = x_list[i] - x_list[j]
        h = y_list[i] - y_list[j]
        best = max(best, h * h + w * w)

fout.write(f"{best}\n")
fout.close()        
        
    





        
